modules/fundamental/src/ai/endpoints/test.rs

```diff
@@ -0,0 +1,40 @@
+use crate::ai::model::ChatState;
+use crate::ai::service::test::ingest_fixtures;
+use crate::ai::service::AiService;
+use crate::test::caller;
+use actix_http::StatusCode;
+use actix_web::test::TestRequest;
+use test_context::test_context;
+use test_log::test;
+use trustify_test_context::call::CallService;
+use trustify_test_context::TrustifyContext;
+
+#[test_context(TrustifyContext)]
+#[test(actix_web::test)]
+async fn configure(ctx: &TrustifyContext) -> anyhow::Result<()> {
+    let service = AiService::new(ctx.db.clone());
+    if !service.enabled() {
+        return Ok(()); // skip test
+    }
+
+    ingest_fixtures(ctx).await?;
+
+    let app = caller(ctx).await?;
+    let mut req = ChatState::new();
+    req.add_human_message("What is the latest version of Trusted Profile Analyzer?".into());
+
+    let request = TestRequest::post()
+        .uri("/api/v1/ai/completions")
+        .set_json(req)
+        .to_request();
+
+    let response = app.call_service(request).await;
+    log::debug!("Code: {}", response.status());
+    assert_eq!(response.status(), StatusCode::OK);
+
+    let result: ChatState = actix_web::test::read_body_json(response).await;
+    log::info!("result: {:?}", result);
+    assert!(result.messages.last().unwrap().content.contains("37.17.9"));
```

Comment:
At the risk of exposing my AI ignorance, how does this work? Where is the magic that integrates the LLM with the contents of our DB?

Reply:
A list of tools can be added to the AI agent.  The first call to the LLM contains the user question and a list of tools that are available.  The LLM will responds with a tool request if it wants use a tool, the agent then invokes the tool and adds it's results to the chat context. and calls the LLM again and so on.

the tool which is providing that DB info for that test is implemented at:
https://github.com/trustification/trustify/pull/830/files#diff-6b73c5edc3e0873b62eaf3ca01c547ec2b770f03663025856e3e3ae13bf7bc77R54

Reply:
Ah, so the magic is in our implementation of `langchain_rust::tools::Tool`? Makes sense, thanks.